FindIntelMKL.cmake

```diff
@@ -253,6 +253,16 @@ else()
   set( IntelMKL_lp64_FOUND FALSE )
 endif()
 
+# SYCL
+if(ENABLE_DPCPP)
```

Comment:
SYCL bindings =/= DPCPP. The way I'd go about this is adding a `sycl` component to `FindBLAS` rather than having a free-floating variable to manipulate these types of behaviors

Reply:
I tried that initially, could not get the `sycl` component to work. Will re-try and create an issue if it does not work this time.